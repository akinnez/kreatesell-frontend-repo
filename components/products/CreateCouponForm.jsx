import { useFormik } from "formik";
import styles from "./CreateCouponForm.module.scss";
import style from '../../public/css/AllProducts.module.scss'
import {TextInput,
    CustomSelect,
    Switch,
    Button
} from '../../components/inputPack'
import { Radio } from 'antd';
import {useState} from 'react'
// import DateTimePicker from 'react-datetime-picker';
import DateTimePicker from 'react-datetime-picker/dist/entry.nostyle'
import 'react-datetime-picker/dist/DateTimePicker.css'
import 'react-calendar/dist/Calendar.css'
import 'react-clock/dist/Clock.css'

export const CreateCouponForm = () =>{
    const [isPercentage, setIsPercentage] = useState(true)
    const [isAmount, setIsAmount] = useState(false)
    const [isLimited, setIsLimited] = useState(true)
    const [isUnLimited, setIsUnLimited] = useState(false)
    const [isUsage, setIsUsage] = useState(true)
    const [isUsageUnlimited, setIsUsageUnlimited] = useState(false)
    const [isAllProduct, setIsAllProduct] = useState(true)
    const [dateAndTime, setDateAndTime] = useState(new Date())
    const [toDateAndTime, setToDateAndTime] = useState(new Date())
    const initialValues = {
        "coupon_settings": {
            "coupon_code": "",
            "is_coupon": true,
            "start_date": "",
            "end_date": "",
            "fixed_amount_value": 0,
            "percentage_value": 0,
            "is_percentage": true,
            "is_apply_to_recurring": true,
            "is_fixed_amount": true,
            "is_coupon_limited": true,
            "no_of_frequency": 0,
            "is_usage_limited": true,
            "no_of_usage": 0,
            "is_for_all_product": true,
            "product_id": 0
          },
          "action": "string",
          "coupon_id": 0
    }
    const handleSubmit = (values)=>{
        // console.log(values)
        // createCoupon()
    }

    const handleChangeField = (setSelfChecked, setPrevCheck)=>{
        setSelfChecked(true)
        setPrevCheck(false)
        // setFieldValue(`coupon_settings.${is_for_all_product}`, true)
    }
    const formik = useFormik({
		initialValues,
		onSubmit: handleSubmit,
		// validationSchema: "",
		validateOnChange: false,
	})
    const { errors, values, setFieldValue } = formik;
    const {coupon_code, product_id, fixed_amount_value, no_of_usage, no_of_frequency, percentage_value} = values.coupon_settings
    const handleRadioChange = (e)=>{
        // console.log(e.target.value)
        if(e.target.value === true){
            setFieldValue("coupon_settings.is_for_all_product", true)
            setFieldValue("coupon_settings.product_id", 0)
            return
        }
        setFieldValue("coupon_settings.is_for_all_product", false)
        // const {name, value} = e.target
    }
    return(
        <div className={`px-0 lg:px-8 ${styles.container}`}>
            <div className="flex flex-col">
                <h1 className="font-bold text-center text-2xl">Create New Coupon</h1>
                <h3 className={style.lightGrey +" font-semibold text-center text-base"}>Generate a new coupon that drives more sales to your product(s).</h3>
            </div>
            <form onSubmit={formik.handleSubmit} className={styles.formContent + " flex flex-col"}>
                <div className="">
                    <TextInput
                        type="text"
                        label="Coupon Code"
                        labelExtra=" Letters and Numbers only!."
                        style={{ width: "100%" }}
                        placeholder="EX: EMBER100"
                        name="coupon_code" 
                        onChange={(e)=>setFieldValue("coupon_settings.coupon_code", e)}
                        value={coupon_code}
                    />
                </div>
                <div className="flex flex-col mt-8">
                    <h2 className="font-semibold text-base">Apply Coupon to</h2>
                    <div className="flex items-center">
                        <Radio.Group onChange={handleRadioChange} >
                            <Radio className={styles.radioContent} value={true} checked={isAllProduct ? true : false}>All Products</Radio>
                            <Radio className={styles.radioContent} value={false} checked={!isAllProduct ? true : false}>Choose Speific Product</Radio>
                        </Radio.Group>
                    </div>
                    <CustomSelect
                        name="product_id"
                        value={product_id}
                        onChange={(e)=>setFieldValue("coupon_settings.product_id", e)}
                        disabled={isAllProduct ? true : false}
                        list={[{value:"Land of Hope and Opportunities", label: "Land of Hope and Opportunities"}, {value:"Multiple wins", label:"Multiple wins"}]}
                    />
                </div>

                <div className="flex flex-col mt-5">
                    <h2 className="font-semibold text-base">Coupon Type</h2>
                    <div className="flex">
                        <div className="col-3">
                            <Radio className={styles.radioContent} onClick={()=>handleChangeField(setIsPercentage, setIsAmount )} checked={isPercentage ? true : false} value={'1'}>Percentage Off</Radio>
                            <TextInput 
                                type="number"
                                placeholder="0"
                                name="percentage_value"
                                onChange={(e)=>setFieldValue("coupon_settings.percentage_value", e)}
                                disabled={!isPercentage ? true : false}
                                value={percentage_value}
                                />
                                
                        </div>
                        <div className="col-3 ml-8">
                            <Radio className={styles.radioContent} onClick={()=>handleChangeField(setIsAmount, setIsPercentage)} checked={isAmount ? true : false}>Amount Off</Radio>
                            <TextInput 
                                type="number"
                                placeholder="0"
                                name="fixed_amount_value"
                                value={fixed_amount_value}
                                disabled={!isAmount ? true : false} 
                                onChange={(e)=>setFieldValue("coupon_settings.fixed_amount_value", e)}
                                />
                                
                        </div>
                     </div>
                </div>

                <div className="flex items-center mt-8 mb-3">
                    <div className={`pt-1 w-full flex items-center`}>
                        <h2 className={styles.label + " text-lg mb-0 font-normal"}>From</h2>
                        <div className={styles.inputGroup +" w-full"}>
                            <DateTimePicker className="w-3/4" onChange={setDateAndTime} value={dateAndTime} />
                        </div>
                    </div>
                    <div className={`pt-1 w-full flex items-center`}>
                        <h2 className={styles.label + " text-lg mb-0 font-normal"}>To</h2>
                        <div className={styles.inputGroup +" w-full"}>
                            
                            <DateTimePicker className="w-3/4" onChange={setToDateAndTime} value={toDateAndTime} />
                        </div>
                    </div>
                </div>
                <div className="flex flex-col mt-8">
                    <h2 className="font-semib5old text-base">Limit the Frequency of the Coupon</h2>
                    <div className="flex items-center mt-2 mb-4">
                        <div>
                            <Radio className={styles.radioContent} onClick={()=>handleChangeField(setIsUnLimited, setIsLimited)} checked={isUnLimited ? true : false} value={'1'}>Unlimited</Radio>
                        </div>
                        <div className="ml-5">
                            <Radio className={styles.radioContent} onClick={()=>handleChangeField(setIsLimited, setIsUnLimited)} checked={isLimited ? true : false} value={'1'}>Limited</Radio>
                        </div>
                    </div>
                    <TextInput
                        type="number"
                        label="Number of Times"
                        style={{ width: "100%" }}
                        placeholder="1"
                        disabled={isUnLimited ? true : false}
                        name="no_of_frequency"
                        value={no_of_frequency}
                        onChange={(e)=>setFieldValue("coupon_settings.no_of_frequency", e)}
                    />
                </div>
                <div className="flex flex-col mt-8">
                    <h2 className="font-semibold text-base">Limit the Usage per Customer</h2>
                    <div className="flex items-center mb-4">
                        <div>
                            <Radio className={styles.radioContent} onClick={()=>handleChangeField(setIsUsageUnlimited, setIsUsage)} checked={isUsageUnlimited ? true : false} value={'1'}>Unlimited use per Customer</Radio>
                            {/* <Radio value={"not checked"} label="" /> */}
                        </div>
                        <div className="ml-5">
                            <Radio className={styles.radioContent} onClick={()=>handleChangeField(setIsUsage, setIsUsageUnlimited)} checked={isUsage ? true : false} value={'1'}>Coupon can be used how many times by a customer</Radio>
                            {/* <Radio label="" /> */}
                        </div>
                    </div>
                    <TextInput
                        type="number"
                        label="Number of times coupon can be used per customer"
                        style={{ width: "100%" }}
                        placeholder="1"
                        name="no_of_usage"
                        value={no_of_usage}
                        disabled={isUsage ? false : true}
                        onChange={(e)=>setFieldValue("coupon_settings.no_of_usage", e)}
                    />
                </div>
                <div >
                    <Switch className={styles.switchContent} label="Allow Discount to be Applied for Recurring Purchases" />
                </div>
                <Button 
                    className="justify-self-center w-1/3 self-center mt-4"
                    label="Save"
                    type="submit"
                    />
            </form>
        </div>
    )
}

