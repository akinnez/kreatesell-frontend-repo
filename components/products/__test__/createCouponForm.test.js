import renderer from "react-test-renderer";
import { CreateCouponForm } from "../CreateCouponForm";

it("renders a snapshot of <CreateCouponForm />", () => {
  const elem = renderer.create(<CreateCouponForm />).toJSON();
  expect(elem).toMatchSnapshot();
});
