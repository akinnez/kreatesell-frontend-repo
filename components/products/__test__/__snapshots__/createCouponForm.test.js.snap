// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <CreateCouponForm /> 1`] = `
<div
  className="px-0 lg:px-8 container"
>
  <div
    className="flex flex-col"
  >
    <h1
      className="font-bold text-center text-2xl"
    >
      Create New Coupon
    </h1>
    <h3
      className="lightGrey font-semibold text-center text-base"
    >
      Generate a new coupon that drives more sales to your product(s).
    </h3>
  </div>
  <form
    className="formContent flex flex-col"
    onSubmit={[Function]}
  >
    <div
      className=""
    >
      <label
        className="jsx-2372179518"
      >
        Coupon Code
         
        <span
          className="jsx-2372179518"
        >
          - 
           Letters and Numbers only!.
        </span>
      </label>
      <input
        className="jsx-2372179518 "
        name="coupon_code"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="EX: EMBER100"
        style={
          Object {
            "width": "100%",
          }
        }
        type="text"
        value=""
      />
    </div>
    <div
      className="flex flex-col mt-8"
    >
      <h2
        className="font-semibold text-base"
      >
        Apply Coupon to
      </h2>
      <div
        className="flex items-center"
      >
        <div
          className="ant-radio-group ant-radio-group-outline"
        >
          <label
            className="ant-radio-wrapper radioContent"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value={true}
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              All Products
            </span>
          </label>
          <label
            className="ant-radio-wrapper radioContent"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value={false}
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Choose Speific Product
            </span>
          </label>
        </div>
      </div>
      <div
        className="jsx-2101433266 input-wrapper"
      >
        <div
          className="jsx-2101433266 input-plus-label-wrapper"
        >
          <label
            className="jsx-2101433266"
          />
          <div
            className=" css-14jk2my-container"
            onKeyDown={[Function]}
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-1y30gap-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-14100a9-ValueContainer"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  className="css-1bwsx9e-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={true}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-0"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    fill="none"
                    height="16"
                    viewBox="0 0 11 16"
                    width="11"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0.5 10.5L5.5 15.5L10.5 10.5H0.5Z"
                      fill="#595959"
                    />
                    <path
                      d="M0.5 5.5L5.5 0.5L10.5 5.5H0.5Z"
                      fill="#595959"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <span
            className="jsx-2101433266 "
          />
        </div>
      </div>
    </div>
    <div
      className="flex flex-col mt-5"
    >
      <h2
        className="font-semibold text-base"
      >
        Coupon Type
      </h2>
      <div
        className="flex"
      >
        <div
          className="col-3"
        >
          <label
            className="ant-radio-wrapper ant-radio-wrapper-checked radioContent"
          >
            <span
              className="ant-radio ant-radio-checked"
              style={Object {}}
            >
              <input
                checked={true}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value="1"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Percentage Off
            </span>
          </label>
          <label
            className="jsx-2372179518"
          >
             
          </label>
          <input
            className="jsx-2372179518 "
            disabled={false}
            name="percentage_value"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="0"
            type="number"
            value={0}
          />
        </div>
        <div
          className="col-3 ml-8"
        >
          <label
            className="ant-radio-wrapper radioContent"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Amount Off
            </span>
          </label>
          <label
            className="jsx-2372179518"
          >
             
          </label>
          <input
            className="jsx-2372179518 "
            disabled={true}
            name="fixed_amount_value"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="0"
            type="number"
            value={0}
          />
        </div>
      </div>
    </div>
    <div
      className="flex items-center mt-8 mb-3"
    >
      <div
        className="pt-1 w-full flex items-center"
      >
        <h2
          className="label text-lg mb-0 font-normal"
        >
          From
        </h2>
        <div
          className="inputGroup w-full"
        >
          <div
            className="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled w-3/4"
            onFocus={[Function]}
          >
            <div
              className="react-datetime-picker__wrapper"
            >
              <div
                className="react-datetime-picker__inputGroup"
                onClick={[Function]}
              >
                <input
                  max="275760-09-13T01:00"
                  min="0001-01-01T00:00"
                  name="datetime"
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={60}
                  style={
                    Object {
                      "position": "absolute",
                      "visibility": "hidden",
                      "zIndex": "-999",
                    }
                  }
                  type="datetime-local"
                  value="2022-04-25T14:28"
                />
                
                <input
                  autoComplete="off"
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                  data-input="true"
                  inputMode="numeric"
                  max={12}
                  min={1}
                  name="month"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="4"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                  data-input="true"
                  inputMode="numeric"
                  max={30}
                  min={1}
                  name="day"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="25"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                  data-input="true"
                  inputMode="numeric"
                  max={275760}
                  min={1}
                  name="year"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="----"
                  required={null}
                  step={1}
                  type="number"
                  value="2022"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                   
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                  data-input="true"
                  inputMode="numeric"
                  max={12}
                  min={1}
                  name="hour12"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="2"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  :
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute"
                  data-input="true"
                  inputMode="numeric"
                  max={59}
                  min={0}
                  name="minute"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="28"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                   
                </span>
                <select
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                  data-input="true"
                  data-select="true"
                  name="amPm"
                  onChange={[Function]}
                  required={null}
                  value="pm"
                >
                  <option
                    value="am"
                  >
                    AM
                  </option>
                  <option
                    value="pm"
                  >
                    PM
                  </option>
                </select>
                
              </div>
              <button
                className="react-datetime-picker__clear-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                className="react-datetime-picker__calendar-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pt-1 w-full flex items-center"
      >
        <h2
          className="label text-lg mb-0 font-normal"
        >
          To
        </h2>
        <div
          className="inputGroup w-full"
        >
          <div
            className="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled w-3/4"
            onFocus={[Function]}
          >
            <div
              className="react-datetime-picker__wrapper"
            >
              <div
                className="react-datetime-picker__inputGroup"
                onClick={[Function]}
              >
                <input
                  max="275760-09-13T01:00"
                  min="0001-01-01T00:00"
                  name="datetime"
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={60}
                  style={
                    Object {
                      "position": "absolute",
                      "visibility": "hidden",
                      "zIndex": "-999",
                    }
                  }
                  type="datetime-local"
                  value="2022-04-25T14:28"
                />
                
                <input
                  autoComplete="off"
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                  data-input="true"
                  inputMode="numeric"
                  max={12}
                  min={1}
                  name="month"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="4"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                  data-input="true"
                  inputMode="numeric"
                  max={30}
                  min={1}
                  name="day"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="25"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                  data-input="true"
                  inputMode="numeric"
                  max={275760}
                  min={1}
                  name="year"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="----"
                  required={null}
                  step={1}
                  type="number"
                  value="2022"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                   
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                  data-input="true"
                  inputMode="numeric"
                  max={12}
                  min={1}
                  name="hour12"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="2"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  :
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute"
                  data-input="true"
                  inputMode="numeric"
                  max={59}
                  min={0}
                  name="minute"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="28"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                   
                </span>
                <select
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                  data-input="true"
                  data-select="true"
                  name="amPm"
                  onChange={[Function]}
                  required={null}
                  value="pm"
                >
                  <option
                    value="am"
                  >
                    AM
                  </option>
                  <option
                    value="pm"
                  >
                    PM
                  </option>
                </select>
                
              </div>
              <button
                className="react-datetime-picker__clear-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                className="react-datetime-picker__calendar-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col mt-8"
    >
      <h2
        className="font-semib5old text-base"
      >
        Limit the Frequency of the Coupon
      </h2>
      <div
        className="flex items-center mt-2 mb-4"
      >
        <div>
          <label
            className="ant-radio-wrapper radioContent"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value="1"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Unlimited
            </span>
          </label>
        </div>
        <div
          className="ml-5"
        >
          <label
            className="ant-radio-wrapper ant-radio-wrapper-checked radioContent"
          >
            <span
              className="ant-radio ant-radio-checked"
              style={Object {}}
            >
              <input
                checked={true}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value="1"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Limited
            </span>
          </label>
        </div>
      </div>
      <label
        className="jsx-2372179518"
      >
        Number of Times
         
      </label>
      <input
        className="jsx-2372179518 "
        disabled={false}
        name="no_of_frequency"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="1"
        style={
          Object {
            "width": "100%",
          }
        }
        type="number"
        value={0}
      />
    </div>
    <div
      className="flex flex-col mt-8"
    >
      <h2
        className="font-semibold text-base"
      >
        Limit the Usage per Customer
      </h2>
      <div
        className="flex items-center mb-4"
      >
        <div>
          <label
            className="ant-radio-wrapper radioContent"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value="1"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Unlimited use per Customer
            </span>
          </label>
        </div>
        <div
          className="ml-5"
        >
          <label
            className="ant-radio-wrapper ant-radio-wrapper-checked radioContent"
          >
            <span
              className="ant-radio ant-radio-checked"
              style={Object {}}
            >
              <input
                checked={true}
                className="ant-radio-input"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="radio"
                value="1"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Coupon can be used how many times by a customer
            </span>
          </label>
        </div>
      </div>
      <label
        className="jsx-2372179518"
      >
        Number of times coupon can be used per customer
         
      </label>
      <input
        className="jsx-2372179518 "
        disabled={false}
        name="no_of_usage"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="1"
        style={
          Object {
            "width": "100%",
          }
        }
        type="number"
        value={0}
      />
    </div>
    <div
      className="switchContent"
    >
      <div
        className="jsx-2729235566 switch-wrapper"
      >
        <span
          className="jsx-2729235566 label"
        >
          Allow Discount to be Applied for Recurring Purchases
        </span>
        <label
          className="jsx-2729235566 switch"
        >
          <input
            className="jsx-2729235566"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="jsx-2729235566 slider round"
          />
        </label>
      </div>
    </div>
    <button
      className="jsx-1869262872 justify-self-center w-1/3 self-center mt-4"
      onClick={[Function]}
      type="submit"
    >
       
      Save
    </button>
  </form>
</div>
`;
